### ISSUE TITLE
[no-unused-vars] false positive on abstract method declaration

### ISSUE THREAD

**Issue by mcalmus** [typescript-eslint/typescript-eslint#2926]
- [n/a] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  }
}
```

```TS
export abstract class Foo {
   abstractfooMethod(a: string, b: string, c: string): string;
}
```

**Expected Result**

Does not produce a lint error

**Actual Result**

```error	'a' is defined but never used	no-unused-vars```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.17.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
Please read the FAQ - https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code
